app/models/game.rb

```diff
@@ -36,6 +36,17 @@ def check?
     kings_in_check?('white') || kings_in_check?('black')
   end
 
+  def forfeit(current_user)
+    if current_user.id == white_player_id
```

Comment:
For this, you don't have to use ID's. ActiveRecord will compare objects for you

`if current_user == white_player`